### PR TITLE
[ISSUE #277] support prometheus http extension to make sdk can be pulled by prometheus

### LINKED ISSUE #277: support prometheus http extension to make sdk can be pulled by prometheus
**What is the feature you want to add?**
add prometheus extension, user can do with the node exporters to get the node metrics

**Why do you want to add this feature?**

**How to implement this feature?**

**Additional context**

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #277 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [x] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: bootstrap/server.go [hunk 1]
```diff
@@ -139,64 +135,128 @@ func StartComponents(ctx context.Context, cfg *config.Config) error {
...
 		return err
 	}
 
 	return nil
 }
 
-// StartConfigModule 启动配置中心模块
-func StartConfigModule(ctx context.Context, cfg *config.Config) error {
+// StartConfigCenterComponents 启动配置中心模块
+func StartConfigCenterComponents(ctx context.Context, cfg *config.Config) error {
```
Comment (andrewshan) [inline bootstrap/server.go:259]: 这里最好在import时候，从别名区分下配置中心的config，以及bootstrap的前缀

配置中心  config_center
启动配置 boot_config

File: cache/cache.go [hunk 2]
```diff
@@ -47,36 +48,106 @@ const (
...
 	// clear
-	//  @return error
 	clear() error
 
 	// name
-	//  @return string
 	name() string
 }
 
+type basCache struct {
```
Comment (andrewshan) [inline cache/cache.go:109]: 这个最好加上注释

File: cache/cache.go [hunk 3]
```diff
@@ -47,36 +48,106 @@ const (
...
+			listeners: make([]Listener, 0, 4),
+		},
+	}
+}
+
+// initialize
+// @param c
+// @return error
+func (bc *basCache) initialize(c map[string]interface{}) error {
+	return errors.New("implement me")
```
Comment (andrewshan) [inline cache/cache.go:126]: 如果用不到的方法，可以去掉

File: cache/client.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+		ret = append(ret, value)
+		return true
+	})
+
+	amount := uint32(len(ret))
+	return amount, doClientPage(ret, offset, limit), nil
+}
+
+// doClientPage 进行分页
+func doClientPage(ret []*model.Client, offset, limit uint32) []*model.Client {
```
Comment (andrewshan) [inline cache/client.go:240]: 这个是控制台查询才会触发的排序是吧？

File: cache/default.go [hunk 5]
```diff
@@ -90,11 +82,26 @@ func initialize(ctx context.Context, cacheOpt *Config, storage store.Store, list
 	cacheMgn.caches[CacheUser] = newUserCache(storage, notify)
 	cacheMgn.caches[CacheAuthStrategy] = newStrategyCache(storage, notify, cacheMgn.caches[CacheUser].(UserCache))
 	cacheMgn.caches[CacheNamespace] = newNamespaceCache(storage)
+	cacheMgn.caches[CacheClient] = newClientCache(storage)
 
 	if err := cacheMgn.initialize(); err != nil {
 		return err
 	}
 
+	cacheMgn.caches[CacheInstance].addListener([]Listener{
```
Comment (andrewshan) [inline cache/default.go:91]: CacheInstance的缓存，代理执行到这里的时候，可以保证不为空吗

File: cache/client.go (hunk 4 above)
Comment (chuntaojun) [reply cache/client.go:240]: 是的，只针对控制台查询

**Pull request merged**